src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -1262,8 +1262,10 @@ security_manager::SSLContext::HandshakeContext
 ApplicationManagerImpl::GetHandshakeContext(uint32_t key) const {
   LOG4CXX_AUTO_TRACE(logger_);
   ApplicationConstSharedPtr app = application(key);
+  if (!app) {
```

Comment:
I would raher re-write this method as 

>security_manager::SSLContext::HandshakeContext res;
if (app) {
    res.make_context(custom_str::CustomString(app->policy_app_id()),  app->name());
}
return res;